pkg/receive/multitsdb.go

```diff
@@ -356,10 +380,12 @@ func (t *MultiTSDB) pruneTSDB(ctx context.Context, logger log.Logger, tenantInst
 		return false, err
 	}
 
-	if err := os.RemoveAll(tenantTSDB.db.Dir()); err != nil {
+	if err := os.RemoveAll(tdb.Dir()); err != nil {
 		return false, err
 	}
 
+	tenantInstance.readyS.set(nil)
```

Comment:
Could we hit an 'unlucky' panic when returning [appender](https://github.com/thanos-io/thanos/blob/main/pkg/receive/multitsdb.go#L583) e.g. in writer, when we have `nil` storage but the tenant was not yet pruned?

Reply:
From what I see we always handle `nil` storage in the codebase. Storage will also be nil when a tenant is not yet initialized, so it doesn't seem to be a new case.

Reply:
You're right, resolving this 👍